apps/E2E/src/Menu/specs/Menu.spec.win.ts

```diff
@@ -57,18 +61,124 @@ describe('Menu Functional Testing', () => {
   it('Validate OnOpenChange() callback was fired -> Click', async () => {
     await MenuPageObject.clickComponent();
     await expect(await MenuPageObject.didMenuOpen()).toBeTruthy();
+
     await expect(await MenuPageObject.didAssertPopup()).toBeFalsy(MenuPageObject.ERRORMESSAGE_ASSERT);
   });
 
   it('Validate OnOpenChange() callback was fired -> Type "Enter"', async () => {
-    await MenuPageObject.sendKey(MenuComponentSelector.PrimaryComponent, Keys.ENTER);
+    await MenuPageObject.sendKey(MenuComponentSelector.MenuTrigger, Keys.ENTER);
     await expect(await MenuPageObject.didMenuOpen()).toBeTruthy();
+
     await expect(await MenuPageObject.didAssertPopup()).toBeFalsy(MenuPageObject.ERRORMESSAGE_ASSERT);
   });
 
   it('Validate OnOpenChange() callback was fired -> Type "SPACE"', async () => {
-    await MenuPageObject.sendKey(MenuComponentSelector.PrimaryComponent, Keys.SPACE);
+    await MenuPageObject.sendKey(MenuComponentSelector.MenuTrigger, Keys.SPACE);
     await expect(await MenuPageObject.didMenuOpen()).toBeTruthy();
+
+    await expect(await MenuPageObject.didAssertPopup()).toBeFalsy(MenuPageObject.ERRORMESSAGE_ASSERT);
+  });
+
+  it('Validate MenuItem onClick() callback fired -> Click', async () => {
+    await MenuPageObject.clickComponent();
```

Comment:
I think we can improve the performance of these tests. It doesn't seem efficient to have to close/re-open the menu in every test - wastes time and resources.

Let's put all the "MenuItem onClick()` tests in one test (ie. click, enter, space). No need to close/open the menu.

Then, in each `expect` statement, you can create a error message specific to each action you performed

In the interest of performance,

Reply:
The tests definitely do take longer than they should. Will optimize.